route_following_plugin/src/route_following_plugin.cpp

```diff
@@ -104,30 +103,47 @@ namespace route_following_plugin
+            ROS_ERROR_STREAM("following_lanelets.size():" << following_lanelets.size());
+
             double target_speed_in_lanelet = findSpeedLimit(route_shortest_path[shortest_path_index]);
 
             //update start distance and start speed from previous maneuver if it exists
             start_dist = (maneuvers.empty()) ? wm_->routeTrackPos(route_shortest_path[shortest_path_index].centerline2d().front()).downtrack : GET_MANEUVER_PROPERTY(maneuvers.back(), end_dist);
             start_speed = (maneuvers.empty()) ? 0.0 : GET_MANEUVER_PROPERTY(maneuvers.back(), end_speed);
+            ROS_DEBUG_STREAM("start_dist:" << start_dist << ", start_speed:" << start_speed);
+            ROS_ERROR_STREAM("start_dist:" << start_dist << ", start_speed:" << start_speed);
 
+                
             end_dist = wm_->routeTrackPos(route_shortest_path[shortest_path_index].centerline2d().back()).downtrack;
+            ROS_DEBUG_STREAM("end_dist:" << end_dist);
+            ROS_ERROR_STREAM("end_dist:" << end_dist);
+
             end_dist = std::min(end_dist, route_length);
-            
-            if (std::fabs(start_dist - end_dist) < 0.1) //TODO: edge case that was not recreatable. Sometimes start and end dist was same which crashes inlanecruising
-            {
-                ROS_WARN_STREAM("start and end dist are equal! shortest path id" << shortest_path_index << ", lanelet id:" << route_shortest_path[shortest_path_index].id() <<
-                    ", start and end dist:" << start_dist);
-                continue;
-            }
-            
+            ROS_DEBUG_STREAM("min end_dist:" << end_dist);
+            ROS_ERROR_STREAM("min end_dist:" << end_dist);
+
             if (isLaneChangeNeeded(following_lanelets, route_shortest_path[shortest_path_index + 1].id()))
             {
+                ROS_DEBUG_STREAM("LaneCHangeNeeded");
+                ROS_ERROR_STREAM("LaneCHangeNeeded");
+    
                 maneuvers.push_back(composeLaneChangeManeuverMessage(start_dist, end_dist, start_speed, target_speed_in_lanelet, route_shortest_path[shortest_path_index].id(), route_shortest_path[shortest_path_index + 1].id()));
                 ++shortest_path_index; //Since lane change covers 2 lanelets - skip planning for the next lanelet
+
+                //Determine the Lane Change Status
+                ROS_DEBUG_STREAM("Recording lanechange start_dist <<" << start_dist  << ", from llt id:" << route_shortest_path[shortest_path_index].id() << " to llt id: " << 
+                    route_shortest_path[shortest_path_index+ 1].id());
+                upcoming_lane_change_status_msg_map_.push({start_dist, ComposeLaneChangeStatus(route_shortest_path[shortest_path_index],route_shortest_path[shortest_path_index + 1])});
             }
             else
             {
+                ROS_DEBUG_STREAM("Lanechange NOT Needed ");
+                ROS_ERROR_STREAM("Lanechange NOT Needed :");
```

Comment:
All these duplicate ERROR logs should be removed as they will clog up our log files and are not in fact errors. The debug logs are enough as they can be trivially enabled in carma-config when needed.